<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>338</width>
    <height>632</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>338</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>338</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>CAN-TS Demo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <widget class="QGroupBox" name="gbCanTs">
      <property name="title">
       <string>CAN-TS Settings</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="leCanTsLocalAddress">
         <property name="text">
          <string>Local ID (0x):</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="txtCanTsLocalAddress">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>AA</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="leCanTsRemoteAddress">
         <property name="text">
          <string>Remote ID (0x):</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="txtCanTsRemoteAddress">
         <property name="text">
          <string>80</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="leCanTsTimeout">
         <property name="text">
          <string>Timeout (ms):</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="txtCanTsTimeout">
         <property name="text">
          <string>1000</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1" colspan="2">
     <widget class="QGroupBox" name="gbCanBus">
      <property name="title">
       <string>Serial Port Settings</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="leNominalBus">
         <property name="text">
          <string>Bus 0 (NOM):</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="txtNominalBus">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>COM26</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="leRedundantBus">
         <property name="text">
          <string>Bus 1 (RED):</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="txtRedundantBus">
         <property name="text">
          <string>COM25</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="leBaudRate">
         <property name="text">
          <string>Baudrate:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="txtBaudRate">
         <property name="text">
          <string>500000</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0" colspan="3">
     <widget class="QGroupBox" name="gbKeepAlive">
      <property name="title">
       <string>Keep Alive</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QPushButton" name="btnKeepAliveStart">
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnKeepAliveStop">
         <property name="text">
          <string>Stop</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout5">
         <item>
          <widget class="QLabel" name="leKeepAlivePeriod">
           <property name="text">
            <string>Period (ms):</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="txtKeepAlivePeriod">
           <property name="text">
            <string>1000</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="0" colspan="3">
     <widget class="QGroupBox" name="gbTcTmTransfer">
      <property name="title">
       <string>TC/TM Transfer</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QPushButton" name="btnLedOn">
         <property name="text">
          <string>LED On</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnLedOff">
         <property name="text">
          <string>LED Off</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout2">
         <item>
          <widget class="QLabel" name="leLedStatus">
           <property name="text">
            <string>Status:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lblLedStatus">
           <property name="text">
            <string>LED Off</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="4" column="0" colspan="3">
     <widget class="QGroupBox" name="gbBlockTransfer">
      <property name="title">
       <string>Block Transfer</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Data (0x):</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="2">
        <widget class="QLineEdit" name="txtBlockAddress">
         <property name="text">
          <string>0</string>
         </property>
         <property name="placeholderText">
          <string>From 0 to 255</string>
         </property>
         <property name="clearButtonEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="1" colspan="2">
        <widget class="QPlainTextEdit" name="txtBlockData">
         <property name="plainText">
          <string>123456789ABCDEF0</string>
         </property>
         <property name="placeholderText">
          <string>Up to 256 bytes</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QPushButton" name="btnBlockTransferReceive">
         <property name="text">
          <string>Receive</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="btnBlockTransferSend">
         <property name="text">
          <string>Send</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Address:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string># Frames:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1" colspan="2">
        <widget class="QLineEdit" name="txtNumFrames">
         <property name="text">
          <string>1</string>
         </property>
         <property name="placeholderText">
          <string>Up to 32 frames</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="5" column="0" colspan="3">
     <widget class="QGroupBox" name="gbTimeSync">
      <property name="title">
       <string>Time Sync</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QPushButton" name="btnTimeSyncSend">
         <property name="text">
          <string>Send</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout4">
         <item>
          <widget class="QLabel" name="leTimeSyncTimestamp">
           <property name="text">
            <string>Timestamp (0x):</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="txtTimeSyncTimestamp">
           <property name="text">
            <string>1A2B3C4D5E6F7081</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Communication</string>
      </property>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QPushButton" name="btnOpenPort">
         <property name="minimumSize">
          <size>
           <width>75</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>75</width>
           <height>23</height>
          </size>
         </property>
         <property name="text">
          <string>Open</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="lblConnStatus">
         <property name="text">
          <string>Closed</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="btnClosePort">
         <property name="minimumSize">
          <size>
           <width>75</width>
           <height>23</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>75</width>
           <height>23</height>
          </size>
         </property>
         <property name="text">
          <string>Close</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>STATUS</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1" colspan="2">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>CAN Bus</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Active Bus:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="lblActiveBus">
         <property name="text">
          <string>BUS 0 (N)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="btnSwitchBus">
         <property name="text">
          <string>Switch Bus</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>338</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
